Develop_Commit_1

oppplk1 opened the pull request:
Develop_Commit_1

oppplk1 committed:
Develop_Commit_1

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,15 @@
+def unique_courier():
+    courier_data = generate_unique_courier()
+    response = CourierRoutes.create_courier(courier_data)
+    assert response.status_code == 201, "Не удалось создать курьера"
```
Нужно исправить: фикстуры не тесты, в них assert не используется

AlexKitIt commented on constants.py:
```diff
@@ -0,0 +1,4 @@
+BASE_URL = "https://qa-scooter.praktikum-services.ru"
+COURIER_LOGIN_ENDPOINT = "/api/v1/courier/login"
+ORDER_CREATE_ENDPOINT = "/api/v1/orders"
+EXPECTED_SUCCESS_RESPONSE = {"ok": True}
```
Тексты ответов стоит отделить от эндпоинтов в разные классы

AlexKitIt commented on helpers.py:
```diff
@@ -0,0 +1,26 @@
+        "firstName": first_name
+    }
+
+    response = requests.post("https://qa-scooter.praktikum-services.ru/api/v1/courier", json=payload)
```
Нужно исправить: здесь и далее:  все урлы уже вынесены в константы

AlexKitIt commented on routes/courier_routes.py:
```diff
@@ -0,0 +1,22 @@
+    COURIER_LOGIN = "/api/v1/courier/login"
+    COURIER_DELETE = "/api/v1/courier/{}"
+
+    @classmethod
```
А почему именно классовые методы? Тут статикам место

AlexKitIt commented on tests/test_accept_order.py:
```diff
@@ -0,0 +1,67 @@
+    def test_accept_order_success(self):
+        # Создаём курьера
+        courier_data = generate_unique_courier()
+        assert courier_data, "Не удалось создать курьера"
```
Нужно исправить: здесь и далее: каждый шаг в тестах не ассертится, assert пишется на ключевой проверяемой функциональности

AlexKitIt commented on tests/test_accept_order.py:
```diff
@@ -0,0 +1,67 @@
+
+        # Создаём заказ
+        order_data = {
+            "firstName": "Иван",
+            "lastName": "Иванов",
+            "address": "Ленина 10",
+            "metroStation": 4,
+            "phone": "+7 999 111 22 33",
+            "rentTime": 2,
+            "deliveryDate": "2025-01-01",
+            "comment": "Тест",
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще, в тестах и тестовых классах их держать не стоит

AlexKitIt commented on tests/test_accept_order.py:
```diff
@@ -0,0 +1,67 @@
+    @allure.description("Создаём курьера и заказ, затем курьер принимает заказ.")
+    def test_accept_order_success(self):
+        # Создаём курьера
+        courier_data = generate_unique_courier()
+        assert courier_data, "Не удалось создать курьера"
+        login, password, _ = courier_data
+        login_response = CourierRoutes.login_courier({"login": login, "password": password})
+        assert login_response.status_code == 200, "Курьер не смог авторизоваться"
+        courier_id = login_response.json().get("id")
```
Нужно исправить: здесь и длаее: это все предусловия теста, их не должно быть в тестах, они должны быть в фикстурах

AlexKitIt commented on tests/test_accept_order.py:
```diff
@@ -0,0 +1,67 @@
+    @allure.description("Курьер с ID 999999 пытается принять несуществующий заказ.")
+    def test_accept_order_invalid_courier(self):
+        response = OrderRoutes.accept_order(999999, 123456)
+        assert response.status_code == 404, "Должна быть ошибка при принятии заказа несуществующим курьером"
```
Нужно исправить: здесь и далее: одной проверки на статус код недостаточно, необходимо парсить ответ

oppplk1 committed:
Develop_Commit_2
- assert удалены из фикстур
- урлы вынесены в константы
- классовые методы заменены на статики
- убраны assert лишних шагов в тестах
- тестовые данные вынесены в data.py
- добавлен парсинг ответов и переменные с текстами ошибок в data.py

AlexKitIt commented on routes/courier_routes.py:
```diff
@@ -0,0 +1,22 @@
+
+
+class CourierRoutes(BaseRoutes):
+    COURIER_CREATE = "/api/v1/courier"
+    COURIER_LOGIN = "/api/v1/courier/login"
+    COURIER_DELETE = "/api/v1/courier/{}"
```
Зачем мы это все дублируем если это описано в константах?
